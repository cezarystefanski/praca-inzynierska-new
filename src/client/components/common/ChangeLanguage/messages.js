export default {
  ChangeLanguage: {
    id: 'common--change_language--change',
  },
  CurrentLanguage: {
    id: 'common--change_language--current_language',
  },
};
