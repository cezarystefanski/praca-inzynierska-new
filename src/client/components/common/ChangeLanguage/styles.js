import { withStyles } from 'styles/themes/ExampleTheme/ExampleTheme';

export default withStyles(({ color, font }) => ({
  changeLanguageBlock: {
    display: 'flex',
  },
  container: {
    color: color.white,
    display: 'flex',
    fontSize: font.sizes.small,
  },
  currentLanguageBlock: {
    marginRight: '1.5rem',
  },
  currentLanguageLabel: {
    marginRight: '0.5rem',
  },
  currentLanguageValue: {
    fontWeight: font.weights.bold,
  },
}));
