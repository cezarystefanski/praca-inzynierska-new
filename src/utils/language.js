import {
  LOCAL_STORAGE_PREFIX,
  LOCAL_STORAGE_LANGUAGE,
} from 'consts/names';
import { DEFAULT_LANGUAGE } from 'consts/language';

const languageStorage = `${LOCAL_STORAGE_PREFIX}${LOCAL_STORAGE_LANGUAGE}`;

export const getCurrentLanguage = () => localStorage.getItem(languageStorage);

export const getLocale = () => {
  const localStorageLanguage = getCurrentLanguage();

  if (!localStorageLanguage) {
    const currentLanguage = navigator.language || DEFAULT_LANGUAGE;

    localStorage.setItem(languageStorage, currentLanguage);

    return currentLanguage;
  }

  return localStorageLanguage;
};

export const setLocale = (language) => {
  localStorage.setItem(languageStorage, language);

  document.location.reload();
  return true;
};
