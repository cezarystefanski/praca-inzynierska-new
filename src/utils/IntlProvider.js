import React from 'react';
import PropTypes from 'prop-types';

import { IntlProvider as StockProvider, addLocaleData } from 'react-intl';
import * as en from 'react-intl/locale-data/en';
import * as pl from 'react-intl/locale-data/pl';

import { DEFAULT_LANGUAGE } from 'consts/language';

import messages from 'messages/messages.json';

/* istanbul ignore next line */
if (process.env.NODE_ENV !== 'test') {
  addLocaleData([...en, ...pl]);
}

export const IntlProvider = ({ children, currentLanguage, locale }) => (
  <StockProvider
    locale={locale}
    messages={messages[currentLanguage]}
  >
    {children}
  </StockProvider>
);

IntlProvider.propTypes = {
  children: PropTypes.node,
  currentLanguage: PropTypes.string,
  locale: PropTypes.string,
};

IntlProvider.defaultProps = {
  children: null,
  currentLanguage: DEFAULT_LANGUAGE,
  locale: DEFAULT_LANGUAGE,
};
