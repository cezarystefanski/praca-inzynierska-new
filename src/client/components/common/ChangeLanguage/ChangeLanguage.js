import React from 'react';
import { FormattedMessage } from 'react-intl';

import { css } from 'styles/themes/ExampleTheme/ExampleTheme';

import { getCurrentLanguage } from 'utils/language';
import { LANG_LIST } from 'consts/language';

import messages from './messages';
import withStyles from './styles';

import { LanguageList } from './LanguageList';

export const ChangeLanguageView = ({ styles }) => {
  const currentLanguage = getCurrentLanguage();

  return (
    <div {...css(styles.container)}>
      <FormattedMessage {...messages.CurrentLanguage}>
        {
          text => (
            <div {...css(styles.currentLanguageBlock)}>
              <span {...css(styles.currentLanguageLabel)}>{text}</span>
              <span {...css(styles.currentLanguageValue)}>{LANG_LIST[currentLanguage]}</span>
            </div>
          )
        }
      </FormattedMessage>
      <FormattedMessage {...messages.ChangeLanguage}>
        {
          text => (
            <div {...css(styles.changeLanguageBlock)}>
              <span>{text}</span>
              <LanguageList
                currentLanguage={currentLanguage}
              />
            </div>
          )
        }
      </FormattedMessage>
    </div>
  );
};

export const ChangeLanguage = withStyles(ChangeLanguageView);
