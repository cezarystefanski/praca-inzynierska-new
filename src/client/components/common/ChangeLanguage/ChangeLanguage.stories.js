import React from 'react';
import { storiesOf } from '@storybook/react';

import { IntlProvider } from 'utils/IntlProvider';

import ChangeLanguage from './index';

const stories = storiesOf('common/ChangeLanguage', module);

stories
  .add('without props', () => (
    <IntlProvider locale="en" currentLanguage="en">
      <div style={{ backgroundColor: '#000' }}>
        <ChangeLanguage />
      </div>
    </IntlProvider>
  ));
