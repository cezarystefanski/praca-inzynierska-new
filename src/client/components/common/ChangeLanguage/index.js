import { compose } from 'recompose';

import { ChangeLanguageView as ChangeLanguage } from './ChangeLanguage';

import withStyles from './styles';

const enhance = compose(
  withStyles,
);

const Component = enhance(ChangeLanguage);

export default Component;
