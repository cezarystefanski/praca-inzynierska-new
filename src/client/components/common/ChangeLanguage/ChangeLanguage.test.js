import React from 'react';
import { mount, shallow } from 'enzyme';
import { IntlProvider } from 'utils/IntlProvider';

import ChangeLanguage from './index';

describe('component common/ChangeLanguage', () => {
  const wrapper = mount(
    <IntlProvider>
      <ChangeLanguage />
    </IntlProvider>,
  );

  it('should render correctly', () => {
    expect(wrapper.find('button'))
      .toHaveLength(2);
  });

  it('should not differ from earlier version', () => {
    const shallowWrapper = shallow(
      <ChangeLanguage />,
    ).shallow();

    expect(shallowWrapper)
      .toMatchSnapshot();
  });
});
