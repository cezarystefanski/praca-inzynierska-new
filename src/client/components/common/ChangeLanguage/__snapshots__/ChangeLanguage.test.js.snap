// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component common/ChangeLanguage should not differ from earlier version 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChangeLanguageView
    css={[Function]}
    styles={
      Object {
        "changeLanguageBlock": Object {
          "_definition": Object {
            "display": "flex",
          },
          "_len": 18,
          "_name": "changeLanguageBlock_jro6t0",
        },
        "container": Object {
          "_definition": Object {
            "color": "#fff",
            "display": "flex",
            "fontSize": "12px",
          },
          "_len": 51,
          "_name": "container_c7r3ip",
        },
        "currentLanguageBlock": Object {
          "_definition": Object {
            "marginRight": "1.5rem",
          },
          "_len": 24,
          "_name": "currentLanguageBlock_y4samv",
        },
        "currentLanguageLabel": Object {
          "_definition": Object {
            "marginRight": "0.5rem",
          },
          "_len": 24,
          "_name": "currentLanguageLabel_udgkk6",
        },
        "currentLanguageValue": Object {
          "_definition": Object {
            "fontWeight": 700,
          },
          "_len": 18,
          "_name": "currentLanguageValue_u5j6yl",
        },
        "inactiveLanguage": Object {
          "_definition": Object {
            "color": "#aaa",
            "cursor": "not-allowed",
            "textDecoration": "none",
          },
          "_len": 63,
          "_name": "inactiveLanguage_17eoe3d",
        },
        "languageElement": Object {
          "_definition": Object {
            ":not(:first-child)": Object {
              ":before": Object {
                "color": "#fff",
                "content": "'|'",
                "marginRight": "0.25rem",
              },
            },
            "cursor": "pointer",
            "marginLeft": "0.25rem",
            "textDecoration": "underline",
          },
          "_len": 162,
          "_name": "languageElement_131nfzv",
        },
        "languageList": Object {
          "_definition": Object {
            "display": "flex",
            "listStyle": "none",
            "margin": 0,
            "padding": 0,
          },
          "_len": 60,
          "_name": "languageList_mn4u4y",
        },
        "resettedButton": Object {
          "_definition": Object {
            "background": "none",
            "border": "none",
            "color": "inherit",
            "cursor": "inherit",
            "font": "inherit",
            "outline": "inherit",
            "padding": "0",
            "textDecoration": "inherit",
          },
          "_len": 152,
          "_name": "resettedButton_1wm8fde",
        },
      }
    }
    theme={
      Object {
        "color": Object {
          "grey": "#aaa",
          "icyBlue": "#ccf",
          "white": "#fff",
        },
        "font": Object {
          "sizes": Object {
            "big": "20px",
            "normal": "14px",
            "small": "12px",
          },
          "weights": Object {
            "bold": 700,
            "normal": 500,
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FormattedMessage
          id="common--change_language--current_language"
          values={Object {}}
        >
          [Function]
        </FormattedMessage>,
        <FormattedMessage
          id="common--change_language--change"
          values={Object {}}
        >
          [Function]
        </FormattedMessage>,
      ],
      "className": "container_c7r3ip",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": [Function],
          "id": "common--change_language--current_language",
          "values": Object {},
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": [Function],
          "id": "common--change_language--change",
          "values": Object {},
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormattedMessage
            id="common--change_language--current_language"
            values={Object {}}
          >
            [Function]
          </FormattedMessage>,
          <FormattedMessage
            id="common--change_language--change"
            values={Object {}}
          >
            [Function]
          </FormattedMessage>,
        ],
        "className": "container_c7r3ip",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": [Function],
            "id": "common--change_language--current_language",
            "values": Object {},
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": [Function],
            "id": "common--change_language--change",
            "values": Object {},
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
